modules/document_repository/php/editfile.class.inc

```diff
@@ -0,0 +1,171 @@
+                ->withStatus(200)
+                ->withHeader("Allow", "POST")
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        json_encode("update")
+                    )
+                );
+        }
+        if ($request->getMethod() == "DELETE") {
+            $this->deleteFile();
+            return (new \LORIS\Http\Response())
+                ->withHeader("Content-Type", "text/plain")
+                ->withStatus(200)
+                ->withHeader("Allow", "DELETE")
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        json_encode("delete")
+                    )
+                );
+        }
+    }
+    /**
+ * Handles the document edit process
+ *
+ * @throws \DatabaseException
+ *
+ * @return void
+ */
+    function editDocFile(): void
+    {
+        $db           = \Database::singleton();
+        $user         = \User::singleton();
+        $updateValues = [
+                         'File_category' => isset($_POST['category']) ?
+                              $_POST['category'] : null,
+                         'For_site'      => isset($_POST['forSite']) ?
+                              $_POST['forSite'] : null,
+                         'Instrument'    => isset($_POST['instrument']) ?
+                              $_POST['instrument'] : null,
+                         'comments'      => isset($_POST['comments']) ?
+                              $_POST['comments'] : null,
+                         'version'       => isset($_POST['version']) ?
+                              $_POST['version'] : null,
+                         'visitLabel'    => isset($_POST['visitLabel']) ?
+                              $_POST['visitLabel'] : null,
+                         'pscid'         => isset($_POST['pscid']) ?
+                              $_POST['pscid'] : null,
+                        ];
+        try {
+            $db->update(
+                'document_repository',
+                $updateValues,
+                ['record_id' => $_POST['id']]
+            );
+        } catch (\DatabaseException $e) {
+            showError("Could not update the file. Please try again!");
+        }
+    }
+    /**
+ * Handles the document edit process
+ *
+ * @throws \DatabaseException
+ *
+ * @return void
+ */
+    function deleteFile(): void
+    {
+        $user = \User::singleton();
+        if (!$user->hasPermission('document_repository_delete')) {
+            header("HTTP/1.1 403 Forbidden");
+            exit;
+        }
+
+        // create Database object
+        $DB = \Database::singleton();
+
+        $Notifier = new \NDB_Notifier(
+            "document_repository",
+            "delete"
+        );
+
+        $rid = basename($_SERVER['REQUEST_URI']);
+
+        $fileName = $DB->pselectOne(
+            "Select File_name from document_repository
+             where record_id =:identifier",
```

Comment:
while you're here can you clean up the formatting of these SQL statements? SQL keywords like SELECT, FROM, and WHERE should be upper case.